garden-backend-service/src/api/routes/_utils.py

```diff
@@ -44,3 +54,85 @@ async def is_doi_registered(doi: str) -> bool:
         return True
     else:
         return False
+
+
+async def get_gardens_for_entrypoint(
+    entrypoint: Entrypoint, db: AsyncSession
+) -> list[Garden] | None:
+    garden_entrys = await db.execute(
+        select(Garden)
+        .join(gardens_entrypoints, Garden.id == gardens_entrypoints.c.garden_id)
+        .where(gardens_entrypoints.c.entrypoint_id == entrypoint.id)
+    )
+    return garden_entrys.all()
```

Comment:
I don't think this is actually a `list[Garden]` -- [docs](https://docs.sqlalchemy.org/en/20/core/connections.html#sqlalchemy.engine.Result.all) say this is a sequence of `Row` objects, not `Garden` models. I think using `db.scalars(...` instead of `.execute` here would get us what you intended

Reply:
though it's kinda fun that duck typing let us get away without getting bit here